backend/auth/api_key_handler.py

```diff
@@ -13,14 +13,8 @@ async def verify_api_key(
         # Use UTC time to avoid timezone issues
         current_date = datetime.utcnow().date()
         commons = common_dependencies()
-        result = (
-            commons["supabase"]
-            .table("api_keys")
-            .select("api_key", "creation_time")
-            .filter("api_key", "eq", api_key)
-            .filter("is_active", "eq", True)
-            .execute()
-        )
+        result = commons["db"].get_active_api_key(api_key)
```

Comment:
We follow the use of `commons` here, but I am not sure pushing the db object in a dict enables the VSCode python server to be helpful and know the type of the variable?

Reply:
I agree. I suspect this refactoring should be on a PR of its own